### PR TITLE
Add elixir-ls as a language backend for elixir

### PR DESCRIPTION
**Pull request by mpanarin**
This pull request adds `lsp` as a backend for elixir layer. Alchemist is great, but it is no longer maintained.

Implementation mimics the python lsp backend implementation.

### CODE REVIEW

File: layers/+lang/elixir/funcs.el [hunk 1]
```diff
@@ -9,6 +9,62 @@
...
+

+;;lsp
+
+(defun spacemacs//elixir-setup-lsp ()
+  "Setup lsp backend."
+  (if (configuration-layer/layer-used-p 'lsp)
+      (progn (add-to-list 'exec-path elixir-ls-path) (lsp))
+    (message "`lsp' layer is not installed, please add `lsp' layer to your dotfile."))
+  (if (configuration-layer/layer-used-p 'dap)
```
Comment (yyoncho) [inline layers/+lang/elixir/funcs.el]: seems like this is missing from the readme?

File: layers/+lang/elixir/README.org [hunk 2]
```diff
@@ -125,130 +175,148 @@ Emacs won't ask you if the variable is safe whenever an elixir file is opened.
...
+     (setq elixir-ls-path "*path to folder*/elixir-ls/release")
+   #+END_SRC
+   or in the layer definition
+   #+BEGIN_SRC elisp
+     (elixir :variables elixir-backend 'alchemist elixir-ls-path "*path to folder*/elixir-ls/release")
+   #+END_SRC
+
+   by default =elixir-ls-path= is equal to ="~/elixir-ls/release"=
+
 * Key bindings
```
[the remark this answers is not in the document]
Comment (mpanarin) [reply layers/+lang/elixir/README.org]: I don't think so, `lsp` layer is for that

**Comment by mpanarin** [comment]
@yyoncho @axelson updated

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs :purple_heart:, I have cherry picked your commit into develop you can safely delete your branch.

I am merging this trusting that one of you guys did a test with the new LSP server. I just did a code review and this looks all ok so far. For LSP servers I currently rather go for speed than too much testing. If it breaks anything it should be reverted fast.

**Pull request closed without merging**